index.bs

```diff
@@ -779,6 +780,11 @@ To <dfn>initialize WebTransport over HTTP</dfn>, given a {{WebTransport}} object
     1. Let |error| be a newly [=DOMException/created=] {{WebTransportError}} whose
        {{WebTransportErrorOptions/source}} is `"session"`.
     1. [=Cleanup=] |transport| with |error|.
+  1. If [=verify a certificate hash|verifying the certificate hash=] from |connection| against
+     |serverCertificateHashes| returns false or the certificate does not satisfy [=custom certificate requirements=],
+     then abort the remaining steps and [=queue a network task=] with |transport| to run these steps:
+    1. If |transport|.{{[[State]]}} is `"closed"` or `"failed"`, then abort these steps.
+    1. [=Cleanup=] |transport| with a {{SecurityError}}.
```

Comment:
Note that we always have to return a generic "handshake failed" WebTransportError before the handshake succeeds in order to prevent network scanning (see [here](https://w3c.github.io/webtransport/#protocol-security)).